nemo/collections/asr/data/audio_to_text.py

```diff
@@ -133,6 +133,9 @@ def __init__(
         eos_id: Optional[int] = None,
         pad_id: int = 0,
         index_by_file_id: bool = False,
+        shard_strategy: Optional[str] = None,
+        global_rank: Optional[int] = None,
```

Comment:
I guess that's more of a question to @titu1994, but is it possible to use pytorch's `dist.get_rank()` and same for world size? Or a light-weight wrapper function if we need to support some edge cases (and in fact I think we already have such function `utils.get_rank`)? Just feels that this is a global property that never changes and would be much easier to get it from a function whenever it's needed, than pass around through each model and dataloader.

Reply:
I think AppState is a more valid source of "world size" cause it takes a few more cases into account for model parallel. Ignoring that, you first have to check whether distributed is initialized or not, then you can use a method to get world size and local rank ID. It is definitely possible. 

The problem with the global property is the fact that it's global. Originally tarred datasets used such global variables but some domains had case where training is done in multi gpu, you destroy the distributed context and clean up, then eval on single GPU, all in the same run. This caused clash with such cases cause global setting was still set and not cleaned up automatically.

The other way is not via property but global check via testing if pt is under distributed enabled, whether global rank is correct, etc and then return the rank other wise -1. It's higher cost per call though, but it's fine for a init level thing